src/Simulation/EntryPointDriver/Driver.cs

```diff
@@ -13,66 +13,159 @@
+            ImmutableList.Create("--verbose"), false, "Show additional information about the submission.");
 
         /// <summary>
         /// The target option.
         /// </summary>
-        private OptionInfo<string?> TargetOption { get; }
+        private readonly OptionInfo<string?> TargetOption;
+
+        /// <summary>
+        /// The simulator option.
+        /// </summary>
+        private readonly OptionInfo<string> SimulatorOption;
+
+        /// <summary>
+        /// The driver settings.
+        /// </summary>
+        private readonly DriverSettings settings;
+
+        /// <summary>
+        /// All the registered entry points of the program.
+        /// </summary>
+        private readonly IReadOnlyCollection<IEntryPoint> entryPoints;
 
         /// <summary>
         /// Creates a new driver for the entry point.
         /// </summary>
         /// <param name="settings">The driver settings.</param>
-        /// <param name="entryPoint">The entry point.</param>
-        public Driver(DriverSettings settings, IEntryPoint<TIn, TOut> entryPoint)
+        /// <param name="entryPoints">The entry points.</param>
+        public Driver(DriverSettings settings, IEnumerable<IEntryPoint> entryPoints)
         {
             this.settings = settings;
-            this.entryPoint = entryPoint;
 
-            SimulatorOption = new OptionInfo<string>(
-                settings.SimulatorOptionAliases,
-                entryPoint.DefaultSimulatorName,
+            this.SimulatorOption = new OptionInfo<string>(
+                this.settings.SimulatorOptionAliases,
+                this.settings.DefaultSimulatorName,
                 "The name of the simulator to use.",
                 suggestions: new[]
                 {
-                    settings.QuantumSimulatorName,
-                    settings.ToffoliSimulatorName,
-                    settings.ResourcesEstimatorName,
-                    entryPoint.DefaultSimulatorName
+                    this.settings.QuantumSimulatorName,
+                    this.settings.ToffoliSimulatorName,
+                    this.settings.ResourcesEstimatorName,
+                    this.settings.DefaultSimulatorName
                 });
 
             var targetAliases = ImmutableList.Create("--target");
             const string targetDescription = "The target device ID.";
-            TargetOption = string.IsNullOrWhiteSpace(entryPoint.DefaultExecutionTarget)
+            this.TargetOption = string.IsNullOrWhiteSpace(settings.DefaultExecutionTarget)
                 ? new OptionInfo<string?>(targetAliases, targetDescription)
-                : new OptionInfo<string?>(targetAliases, entryPoint.DefaultExecutionTarget, targetDescription);
+                : new OptionInfo<string?>(targetAliases, this.settings.DefaultExecutionTarget, targetDescription);
+
+            this.entryPoints = entryPoints.ToList().AsReadOnly();
```

Comment:
The array passed in by the generated code should already implement `IReadOnlyCollection<IEntryPoint>`. I think you can change the constructor parameter type to that?

Reply:
Looks like that will work.

Reply:
done